### DISCUSSION TITLE
Weird line editing behavior with git bash

### DISCUSSION THREAD

**Discussion by mhutch** [JanDeDobbeleer/oh-my-posh#4336] [Troubleshoot] [answered]
I'm getting really weird line editing behavior with git bash on Windows.

First up, I'm using Windows Terminal with _CaskaydiaCove Nerd Font Mono_ and the built-in catppuchin theme, unmodified. I have oh-my-posh set up identically in both git bash and Powershell. In PowerShell, everything works fine.

But in git bash, when I use the up arrow to recall the previous command, then try to edit it, stuff gets weird. When I use the left arrow to navigate to the start of the command, I have to hit the left arrow key 14 times to cross the wrapping boundary, and then the cursor stops 14 characters before the start of the command. If I navigate back to the end of the command, the cursor moves out 14 characters into free space.

![oh-my-posh-git-bash-line-editing](https://github.com/JanDeDobbeleer/oh-my-posh/assets/183285/641f31ad-a284-4088-b045-b0cfa373ad53)

This has been happening consistently for me (with some variations depending on the prompt state and line length) since I started using oh-my-posh around the start of 2023. I update everything regularly and am current on everything as of writing this post.

Should I report it as a bug? Is there any workaround?

**Comment by JanDeDobbeleer** [comment, marked as answer]
@mhutch I can't reproduce this on bash on macOS, so highly likely this is another git bash issue we can't solve. That shell (due to its nature) is known to have issues as it mimics unix on a Windows filesystem and has quite a few weird behaviours because of it.
